pkg/labelsfilterdynamic/manager.go

```diff
@@ -0,0 +1,323 @@
+		Signal:                         params.Signal,
+		CiliumNetworkPolicy:            params.CiliumNetworkPolicy,
+		CiliumClusterwideNetworkPolicy: params.CiliumClusterWideNetworkPolicies,
+		NetworkPolicy:                  params.NetworkPolicy,
+		EndpointManager:                params.EndpointManager,
+		NamespaceResource:              params.NamespaceResource,
+	}
+
+	params.JobGroup.Add(
+		job.OneShot("dlf-cilium-network-policy-observer", func(ctx context.Context, health cell.Health) (err error) {
+			for ev := range c.CiliumNetworkPolicy.Events(ctx) {
+				switch ev.Kind {
+				case resource.Upsert, resource.Delete:
+					c.Signal.Event(struct{}{})
+				}
+				ev.Done(nil)
+			}
+			return nil
+		}),
+
+		job.OneShot("dlf-cilium-clusterwide-policy-observer", func(ctx context.Context, health cell.Health) (err error) {
+			for ev := range c.CiliumClusterwideNetworkPolicy.Events(ctx) {
+				switch ev.Kind {
+				case resource.Upsert, resource.Delete:
+					c.Signal.Event(struct{}{})
+				}
+				ev.Done(nil)
+			}
+			return nil
+		}),
+
+		job.OneShot("dlf-network-policy-observer", func(ctx context.Context, health cell.Health) (err error) {
+			for ev := range c.NetworkPolicy.Events(ctx) {
+				switch ev.Kind {
+				case resource.Upsert, resource.Delete:
+					c.Signal.Event(struct{}{})
+				}
+				ev.Done(nil)
+			}
+			return nil
+		}),
+
+		job.OneShot("dlf-controller",
+			func(ctx context.Context, health cell.Health) (err error) {
+				if c.initStore(ctx) != nil {
+					health.Degraded("failed to init stores", err)
+					return fmt.Errorf("error creating Dynamic Label Filter resource stores: %w", err)
+				}
+
+				// Trigger initial event
+				c.Signal.Event(struct{}{})
+
+				c.Run(ctx)
+				health.OK("Ready")
+
+				return nil
+			},
+			job.WithRetry(3, &job.ExponentialBackoff{Min: 100 * time.Millisecond, Max: time.Second}),
+			job.WithShutdown()),
+	)
+
+	return c, nil
+}
+
+func (c *controller) initStore(ctx context.Context) (err error) {
+	c.CiliumNetworkPolicyStore, err = c.CiliumNetworkPolicy.Store(ctx)
+	if err != nil {
+		return
+	}
+	c.NetworkPolicyStore, err = c.NetworkPolicy.Store(ctx)
+	if err != nil {
+		return
+	}
+	c.CiliumClusterwideNetworkPolicyStore, err = c.CiliumClusterwideNetworkPolicy.Store(ctx)
+	if err != nil {
+		return
+	}
+	c.NamespaceResourceStore, err = c.NamespaceResource.Store(ctx)
+	if err != nil {
+		return
+	}
+
+	return nil
+}
+
+// Run places the Controller into its control loop.
+//
+// When new events trigger a signal the control loop will be evaluated.
+//
+// A cancel of the provided ctx will kill the control loop along with the running
+// informers.
+func (c *controller) Run(ctx context.Context) {
+	var (
+		l = log.WithFields(logrus.Fields{
+			"component": "Controller.Run",
+		})
+	)
+
+	l.Info("Cilium Dynamic Label Filter Controller now running...")
+	for {
+		select {
+		case <-ctx.Done():
+			l.Info("Cilium Dynamic Label Filter Controller shut down")
+			return
+		case <-c.Signal.Signal:
```

Comment:
Should this be rate-limited in some way? There's the signal coalescing if triggers arrive faster than the reconciliation can occur, but in effect (assuming fast trigger rate) the bottleneck is the rate at which reconciliation can happen - it's not clear to me that that's the right choice, given (I believe) this can kick of identity/ep regenerations.